examples/pylab_examples/mri_with_eeg.py

```diff
@@ -1,77 +1,79 @@
-"""
-This now uses the imshow command instead of pcolor which *is much
-faster*
+"""Displays a set of subplots with an MRI image, its intensity histogram and
+some EEG traces.
+
+NB: ones uses the imshow command instead of pcolor which *is much faster*.
 """
 
 from __future__ import division, print_function
 
 import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.cbook as cbook
+import matplotlib.cm as cm
 
-from matplotlib.pyplot import *
 from matplotlib.collections import LineCollection
-import matplotlib.cbook as cbook
-# I use if 1 to break up the different regions of code visually
-
-if 1:   # load the data
-    # data are 256x256 16 bit integers
-    dfile = cbook.get_sample_data('s1045.ima.gz')
-    im = np.fromstring(dfile.read(), np.uint16).astype(float)
-    im.shape = 256, 256
-
-if 1:  # plot the MRI in pcolor
-    subplot(221)
-    imshow(im, cmap=cm.gray)
-    axis('off')
-
-if 1:  # plot the histogram of MRI intensity
-    subplot(222)
-    im = np.ravel(im)
-    im = im[np.nonzero(im)]  # ignore the background
-    im = im/(2.0**15)  # normalize
-    hist(im, 100)
-    xticks([-1, -.5, 0, .5, 1])
-    yticks([])
-    xlabel('intensity')
-    ylabel('MRI density')
-
-if 1:   # plot the EEG
-    # load the data
-
-    numSamples, numRows = 800, 4
-    eegfile = cbook.get_sample_data('eeg.dat', asfileobj=False)
-    print('loading eeg %s' % eegfile)
-    data = np.fromstring(open(eegfile, 'rb').read(), float)
-    data.shape = numSamples, numRows
-    t = 10.0 * np.arange(numSamples, dtype=float)/numSamples
-    ticklocs = []
-    ax = subplot(212)
-    xlim(0, 10)
-    xticks(np.arange(10))
-    dmin = data.min()
-    dmax = data.max()
-    dr = (dmax - dmin)*0.7  # Crowd them a bit.
-    y0 = dmin
-    y1 = (numRows - 1) * dr + dmax
-    ylim(y0, y1)
-
-    segs = []
-    for i in range(numRows):
-        segs.append(np.hstack((t[:, np.newaxis], data[:, i, np.newaxis])))
-        ticklocs.append(i*dr)
-
-    offsets = np.zeros((numRows, 2), dtype=float)
-    offsets[:, 1] = ticklocs
-
-    lines = LineCollection(segs, offsets=offsets,
-                           transOffset=None,
-                           )
-
-    ax.add_collection(lines)
-
-    # set the yticks to use axes coords on the y axis
-    ax.set_yticks(ticklocs)
-    ax.set_yticklabels(['PG3', 'PG5', 'PG7', 'PG9'])
-
-    xlabel('time (s)')
-
-show()
+from matplotlib.ticker import MultipleLocator
+
+fig = plt.figure("MRI_with_EEG")
+
+# Load the MRI data (256x256 16 bit integers)
+dfile = cbook.get_sample_data('s1045.ima.gz')
+im = np.fromstring(dfile.read(), np.uint16).astype(float)
+im.shape = (256, 256)
+
+# Plot the MRI image
+ax0 = fig.add_subplot(2, 2, 1)
+ax0.imshow(im, cmap=cm.gray)
+ax0.axis('off')
+
+# Plot the histogram of MRI intensity
+ax1 = fig.add_subplot(2, 2, 2)
+im = np.ravel(im)
+im = im[np.nonzero(im)]  # Ignore the background
+im = im / (2**15)  # Normalize
```

Comment:
I agree that this normalization seemed weird to me too but I did not feel comfortable changing it to (2**16 - 1) as I do not know anything about MRI. PS: anyway, currently there are no units for the MRI...